courses/tasks.py

```diff
@@ -1,15 +1,64 @@
 # -*- coding: utf-8 -*-
+import hashlib
+import hmac
+import json
+import logging
+
+from django.conf import settings
+from django.core.management import call_command
 
 from celery import shared_task
+from microsite_configuration import microsite
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
 
-from django.core.management import call_command
+logger = logging.getLogger(__name__)
 
 
 @shared_task
 def update_courses_meta_data(*args, **kwargs):
     """
-    A task that serves as proxy to the management command.
-    Can be used for instance when a signal is fired to update
-    a single course or to run periodic tasks.
+    A task to call when a course is updated in OpenEdX.
+    It calls:
+    - the "update_courses" management command to update fun-apps' search index.
+    - each course run hook url defined in the `COURSE_HOOKS` settig with scheduling information
     """
+    # Update fun-apps course catalog
     call_command('update_courses', *args, **kwargs)
+
+    hooks = getattr(settings, "COURSE_HOOKS", [])
+    if not hooks:
+        return
+
+    # Synchronize with external course hook
+    course_id = kwargs["course_id"]
+    course_key = CourseKey.from_string(course_id)
+    course = modulestore().get_course(course_key)
+    edxapp_domain = microsite.get_value('site_domain', settings.SITE_NAME)
+    data = {
+        "resource_link": "https://{:s}/courses/{:s}/info/".format(
+            edxapp_domain, course_id
+        ),
+        "start": course.start,
+        "end": course.end,
+        "enrollment_start": course.enrollment_start,
+        "enrollment_end": course.enrollment_end,
+        "languages": ["fr"],
+    }
+
+    for hook in hooks:
+        authorization = hmac.new(
+            hook["secret"].encode("utf-8"),
+            msg=json.dumps(data).encode("utf-8"),
```

Comment:
does the `json.dumps` can be computed outside of the loop ? Doing this, does it avoid unnecessary computing ? Or python is smart enough to cache this instruction ?

Reply:
good point! :pray: